### ISSUE TITLE
Define which methods are needed for bindings

### ISSUE THREAD

**Issue by borrrden** [couchbase/couchbase-lite-core#633]
There are now several dozen (close to 100 I guess) warnings in .NET about methods that are defined in the public header but not exported.  Most of the deal with FLMutable* which I assume can be ignored since we don't use them, but how about these:

`c4doc_createFleeceDoc`
`c4doc_hasOldMetaProperties`
`c4listener_availableAPIs`
`c4listener_start`
`c4listener_free`
~~`FLDump`~~
`FLValue_AsTimestamp`
`FLValue_IsEqual`
`FLValue_Retain`
`FLValue_Release`
`FLArray_IsEmpty`
`FLDictIterator_GetCount`
`FLDictIterator_End`
`FLDeepIterator_*`
`FLSharedKeys_Create`
`FLEncoder_SuppressTrailer`
`FLEncoder_GetNextWritePos`
`FLEncoder_WriteDateString`
`FLEncoder_WriteKeyValue`
`FLEncoder_FinishItem`
`FLEncodeJSONDelta`
`FLEncodeApplyingJSONDelta`
`FLSlice_Copy`
`c4listener_shareDB`
`c4listener_unshareDB`
`c4db_URINameFromPath`
~~`FLDoc_GetData`~~
`FLDoc_GetAllocedData`
~~`FLDumpData`~~
`FLSharedKeys_CreateFromStateData`
`FLSharedKeys_GetStateData`
`FLSharedKeys_Encode`
`FLEncoder_Amend`
`FLEncoder_GetBase`
`FLCreateJSONDelta`
`FLApplyJSONDelta`

The bindings script for .NET will pick up anything in the C/include folder for processing by default, but files can be added for exclusion.

**Comment by borrrden** [comment]
I have updated the above by removing the entries that were simply missing from the windows exports, but present in the unix ones.

**Comment by snej** [comment]
For Couchbase Lite purposes, we don't need any of these — Obj-C and Java statically link LiteCore, and C# must not be calling any of these functions since we're not getting runtime errors (right?)

LiteCore as a general purpose shared library should export all of them, but we're not supporting that use case, and we probably never will since we'll end up defining a Couchbase Lite C API as the public API, and just export that.

So I think we can ignore this. In fact, you could go the other way and _remove_ any exports that aren't needed by either .NET or the LiteCore C unit tests.

**Comment by borrrden** [comment]
I spent a few hours doing this and the result is 11280cfe702cf3f8acdb79f94ec95207d9c6538b.  There is a pretty awful setup for the `LiteCoreREST` target in which it takes both exported Fleece functions from the LiteCore dynamic library, and also from the Fleece static library.  If the definitions files are not *just* right then Windows explodes with multiply defined symbols as it switches to taking a whole load of extra symbols from Fleece static to make up for the ones that are not exported in LiteCore and those then collide.  The list as I committed works on Windows.  Note that there are comments at the beginning and the comment in the middle (`; C4Tests`) is important for the .NET bindings script.  Everything after that point the bindings script will ignore and not generate C# for since it is not needed.  Actually, a good number of exports **before** that line are not technically needed aside from the fact that there are some (seemingly obsolete) LiteCore bindings tests that use them.
